### PR TITLE
Add benchmarking for reindexing Spotlight::Resources

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/models/spotlight/resource.rb [hunk 1]
```diff
@@ -52,16 +45,13 @@ def reindex_with_lock
...
-          blacklight_solr.update params: { commitWithin: 500 },
-                                 data: batch.reject(&:blank?).map { |doc| doc.reverse_merge(existing_solr_doc_hash(doc[unique_key]) || {}) }.to_json,
-                                 headers: { 'Content-Type' => 'application/json' }
+      benchmark "Reindexing #{self} (batch size: #{batch_size})" do
+        run_callbacks :index do
+          documents_to_index.each_slice(batch_size) do |batch|
+            blacklight_solr.update params: { commitWithin: 500 },
+                                   data: batch.to_json,
+                                   headers: { 'Content-Type' => 'application/json' }
+          end
```
Comment (ndushay) [inline app/models/spotlight/resource.rb]: is there a way to find out the HTTP response code?  Would it help?

Comment (cbeer) [inline app/models/spotlight/resource.rb]: I think RSolr throw a big exception if this fails.

Comment (ndushay) [inline app/models/spotlight/resource.rb]: I guess what I'm asking is:  will an error get to the logs?

Comment (ndushay) [inline app/models/spotlight/resource.rb]: looks like it doesn't in gdor either - meh.   moving on.

**Pull request merged**